components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -227,27 +229,60 @@ private void updateGatewayPublishedAPIArtifacts(Connection connection, String ap
     public APIArtifactPropertyValues retrieveAPIArtifactPropertyValues(
             String apiUUID, String envName, String revisionUUId) throws APIManagementException {
 
-        String query = SQLConstants.RETRIEVE_API_ARTIFACT_PROPERTY_VALUES;
+        String query = SQLConstants.RETRIEVE_API_ARTIFACT_PROPERTY_VALUES_WITH_ENV_UUID;
         String organization = null;
         Timestamp deployedTime = null;
+        String envUUID = null;
+        boolean emptyResultSet = true;
         APIArtifactPropertyValues propertyValues = new APIArtifactPropertyValues();
+        Map<String, Environment> readOnlyEnvironments = APIUtil.getReadOnlyEnvironments();
         try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
              PreparedStatement preparedStatement = connection.prepareStatement(query)) {
             preparedStatement.setString(1, apiUUID);
             preparedStatement.setString(2, envName);
             preparedStatement.setString(3, revisionUUId);
             try (ResultSet resultSet = preparedStatement.executeQuery()) {
-                while (resultSet.next()) {
+                if (resultSet.next()) {
```

Comment:
lets keep "while" here. I have read, eading the results partially may results in issues in DB connections/locks

Reply:
fixed